### PR TITLE
DDCE-1186:§9 P15 - Nominee Organisation Summary Page 'Change' Journeys

### PR DESCRIPTION
**Pull request by prameya-phatak**
# DDCE-1186

## New feature 

Nominee Organisation Summary Page 'Change' Journeys

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [x]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [x]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/navigation/NomineesNavigator.scala [hunk 1]
```diff
@@ -260,65 +260,70 @@ class NomineesNavigator @Inject()(implicit frontendAppConfig: FrontendAppConfig)
...
 
     case OrganisationNomineeContactDetailsPage => userAnswers: UserAnswers => userAnswers.get(OrganisationNomineeContactDetailsPage) match {
-      case Some(_) => routes.DeadEndController.onPageLoad() // TODO next page
+      case Some(_) => nomineeRoutes.NomineeDetailsSummaryController.onPageLoad()
       case _ =>  routes.SessionExpiredController.onPageLoad()
     }
 
     case OrganisationNomineeAddressLookupPage => userAnswers: UserAnswers => userAnswers.get(OrganisationNomineeAddressLookupPage) match {
-      case Some(_) => routes.DeadEndController.onPageLoad() // TODO next page
+      case Some(_) => nomineeRoutes.NomineeDetailsSummaryController.onPageLoad()
```
Comment (RafSobol) [inline app/navigation/NomineesNavigator.scala:273]: this should be redirecting to the 'has the address changed' page

File: app/pages/nominees/IsOrganisationNomineeNinoPage.scala [hunk 2]
```diff
@@ -16,12 +16,22 @@
...
 
   override def path: JsPath = NomineeOrganisationIdPage.path \ toString
 
   override def toString: String = "isOrganisationNino"
+
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(true)  => userAnswers.remove(OrganisationAuthorisedPersonPassportPage)
+      case Some(false)  => userAnswers.remove(OrganisationAuthorisedPersonNinoPage)
+      case _ => super.cleanup(value, userAnswers)
```
Comment (RafSobol) [inline app/pages/nominees/IsOrganisationNomineeNinoPage.scala:35]: correct me if I'm wrong, but when cleanup is called, value is always a Some (as if the page is a None, this function won't actually be called) - so you can just replace lines 34 and 35 with a `Case _ => userAnswers.remove(OrganisationAuthorisedPersonNinoPage)`

Comment (prameya-phatak) [reply app/pages/nominees/IsOrganisationNomineeNinoPage.scala]: modified

File: app/navigation/NomineesNavigator.scala (hunk 1 above)
Comment (prameya-phatak) [reply app/navigation/NomineesNavigator.scala]: modified

**Review by RafSobol — APPROVED** [review]

**Pull request merged**